isis/src/base/apps/phocube/main.cpp

```diff
@@ -333,42 +319,38 @@ void IsisMain() {
 
   UpdateBandKey("Width", bb, nvals, "1.0");
 
-
-
   p.EndProcess();
 }
 
-
-//  This propagates the input plane to the output plane, then passes it off to
-//  the general routine
-void phocubeDN(Buffer &in, Buffer &out) {
-  for (int i = 0 ; i < in.size() ; i++) {
-    out[i] = in[i];
-  }
-  phocube(out);
-}
-
-
 //  Computes all the geometric properties for the output buffer.  Certain
 //  knowledge of the buffers size is assumed below, so ensure the buffer
 //  is still of the expected size.
-void phocube(Buffer &out) {
-
-
-  // If the DN option is selected, it is already added by the phocubeDN
-  // function.  We must compute the offset to start at the second band.
-  int skipDN = (dn) ? 64 * 64   :  0;
-
+void phocube(Buffer &in, Buffer &out) {
   for (int i = 0; i < 64; i++) {
     for (int j = 0; j < 64; j++) {
 
       MosData mosd, *p_mosd(0);  // For special mosaic angles
 
-      int index = i * 64 + j + skipDN;
+      int index = i * 64 + j;
+      int inIndex = index;
+
+      if (dn) {
+        out[index] = in[index];
+        index += 64 * 64;
+      }
+
+      // If dn is true, make sure to not overwrite the original pixels with NULL for the dn band
+      if (!specialPixels && IsSpecial(in[inIndex])) {
+        for (int band = (dn) ? 1 : 0; band < nbands; band++) {
+          out[index] = Isis::NULL8;
+          index += 64 * 64;
+        }
+        continue;
+      }
```

Comment:
All Cubes default allocate to NULL pixels. You can skip the band selection logic here by just doing `if (!specialPixels && IsSpecial(in[inIndex])) continue;`

Reply:
Turns out this was a bad suggestion. The cube defaults to Null, but the buffer defaults to 0 and since it never reads from the output image before writing to it, it will fill stuff with 0 instead of Null. Since it would take more time to read from the output cube, it's better to just leave this assignment in.